pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -490,6 +495,72 @@ func generateNetworkEnhancedMetrics(args generateNetworkEnhancedMetricArgs) {
 	})
 }
 
+type generateTmpEnhancedMetricsArgs struct {
+	TmpMax  float64
+	TmpUsed float64
+	Tags    []string
+	Demux   aggregator.Demultiplexer
+	Time    float64
+}
+
+// generateTmpEnhancedMetrics generates enhanced metrics for space used and available in the /tmp directory
+func generateTmpEnhancedMetrics(args generateTmpEnhancedMetricsArgs) {
+	args.Demux.AggregateSample(metrics.MetricSample{
+		Name:       tmpUsedMetric,
+		Value:      args.TmpUsed,
+		Mtype:      metrics.DistributionType,
+		Tags:       args.Tags,
+		SampleRate: 1,
+		Timestamp:  args.Time,
+	})
+	args.Demux.AggregateSample(metrics.MetricSample{
+		Name:       tmpMaxMetric,
+		Value:      args.TmpMax,
+		Mtype:      metrics.DistributionType,
+		Tags:       args.Tags,
+		SampleRate: 1,
+		Timestamp:  args.Time,
+	})
+}
+
+func SendTmpEnhancedMetrics(sendMetrics chan bool, tags []string, metricAgent *ServerlessMetricAgent) {
+	if enhancedMetricsDisabled {
+		return
+	}
+
+	bsize, blocks, bavail, err := statfs(tmpPath)
+	if err != nil {
+		log.Debug("Could not emit tmp enhanced metrics")
+		return
+	}
+	tmpMax := blocks * bsize
+	tmpUsed := bsize * (blocks - bavail)
+
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-sendMetrics:
+			generateTmpEnhancedMetrics(generateTmpEnhancedMetricsArgs{
+				TmpMax:  tmpMax,
+				TmpUsed: tmpUsed,
+				Tags:    tags,
+				Demux:   metricAgent.Demux,
+				Time:    float64(time.Now().UnixNano()) / float64(time.Second),
+			})
+			return
+		case <-ticker.C:
```

Comment:
Just checking my understanding. The reason we're using a ticker here is because we're wanting the tmp_used metric to represent the max amount used over the course of any given invocation?

Reply:
Correct, `tmp_used` represents the peak over the course of the invocation